babel/messages/extract.py

```diff
@@ -136,11 +136,22 @@ def extract_from_dir(dirname=None, method_map=DEFAULT_MAPPING,
         options_map = {}
 
     absname = os.path.abspath(dirname)
+
+    def dir_filter(method_map, dirname):
+        if dirname.startswith('.') or dirname.startswith('_'):
+            return False
+
+        absdir = os.path.join(root, dirname).replace(os.sep, '/')
```

Comment:
Since the logic for ignoring filenames uses

```
filename = relpath(filepath, dirpath)
```

I think this should also use a relative path to the root. Otherwise this might end up ignoring paths that happen to contain an ignored fragment outside the relative root.

That is, if your project lives in `/foobars/myproject/`, and you've ignored `*foobar*` (as it has a special meaning _within_ the `myproject` directory), and you invoke Babel from within `/foobars/myproject/`, this would ignore all files.

Reply:
Thanks for the feedbacks. Just applied your suggestion in https://github.com/python-babel/babel/pull/694/commits/252323ae3452ffcc35f2d19b73a9e83084fc6259